arrow-select/src/coalesce.rs

```diff
@@ -290,6 +323,25 @@ impl BatchCoalescer {
         Ok(())
     }
 
+    /// Returns the number of buffered rows
+    pub fn get_buffered_rows(&self) -> usize {
+        self.buffered_rows
+    }
+
+    /// Push a batch directly to the completed batches
+    pub fn flush_buffer_and_push_batch_to_completed(
```

Comment:
I recommend making this non-pub to keep the API smaller and the code easier to use

```suggestion
    fn flush_buffer_and_push_batch_to_completed(
```

Reply:
Thank you @alamb for good suggestion!